### PR TITLE
page_foreground bugfix

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
Bug found by @PRAVEEN-mnl, where the time was not correct in the page_foreground hdf outputs, this was as the dtype was being set to be a float32 (I don't know why I thought this was a good idea before), but that isn't suitable for gps times

### CODE REVIEW

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -1028,41 +1028,38 @@ def to_coinc_hdf_object(self, file_name):
         # Some fields are special cases
         logging.info("Outputting search results")
         time = self.get_end_time()
-        ofd.create_dataset('time', data=time, dtype=np.float32)
+        ofd['time'] = time
 
         if self._inclusive:
```
Comment (tdent) [inline pycbc/io/hdf.py:1033]: would it be neater to roll both the `if self._inclusive` switches into one?

File: pycbc/io/hdf.py [hunk 2]
```diff
@@ -1028,41 +1028,38 @@ def to_coinc_hdf_object(self, file_name):
...
         fap_exc = self.get_coincfile_array('fap_exc')
-        ofd.create_dataset('p_value_exclusive', data=fap_exc,
-                           dtype=np.float32)
+        ofd['p_value_exclusive'] = fap_exc
 
         # Coinc fields
         for field in ['stat']:
             vals = self.get_coincfile_array(field)
-            ofd.create_dataset(field, data=vals, dtype=np.float32)
+            ofd[field] = vals
```
Comment (tdent) [inline pycbc/io/hdf.py:1050]: I don't think we need to define 'vals' any more, just assign the dataset directly

File: pycbc/io/hdf.py (hunk 1 above)
Comment (GarethCabournDavies) [reply pycbc/io/hdf.py:1033]: fair, will do

File: pycbc/io/hdf.py [hunk 3]
```diff
@@ -1090,20 +1086,18 @@ def to_coinc_hdf_object(self, file_name):
                 vals = vals_valid[ifo][0]
                 valid = vals_valid[ifo][1]
                 vals[np.logical_not(valid)] = -1.
-                ofd.create_dataset(ifo + '_'+field, data=vals,
-                                   dtype=np.float32)
+                ofd[f'{ifo}_{field}'] = vals
```
Comment (tdent) [inline pycbc/io/hdf.py:1089]: maybe good to put a line of comment before the 'vals_valid' lines above to explain what this is intended to do?

File: pycbc/io/hdf.py [hunk 4]
```diff
@@ -1028,41 +1028,38 @@ def to_coinc_hdf_object(self, file_name):
         # Some fields are special cases
         logging.info("Outputting search results")
         time = self.get_end_time()
-        ofd.create_dataset('time', data=time, dtype=np.float32)
+        ofd['time'] = time
```
Comment (tdent) [inline pycbc/io/hdf.py:1031]: as for other fields, no need to define a variable which is only used once

Comment (tdent) [reply pycbc/io/hdf.py:1031]: ok .. maybe put a brief inline comment on that, eg `time will be used to determine active ifos`

**Review by tdent — APPROVED** [review]

File: pycbc/io/hdf.py (hunk 4 above)
Comment (GarethCabournDavies) [reply pycbc/io/hdf.py:1031]:
```suggestion
        # time will be used later to determine active ifos
        ofd['time'] = time
```

**Pull request merged**